workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/BusinessEventRestController.java

```diff
@@ -0,0 +1,417 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.customer;
+
+import com.liferay.client.extension.util.spring.boot3.BaseRestController;
+import com.liferay.customer.constants.ExternalLinkConstants;
+import com.liferay.customer.constants.RoleConstants;
+import com.liferay.customer.service.KoroneikiService;
+import com.liferay.osb.koroneiki.phloem.rest.client.dto.v1_0.ExternalLink;
+import com.liferay.osb.spring.boot.client.zendesk.model.ZendeskOrganization;
+import com.liferay.osb.spring.boot.client.zendesk.model.ZendeskTicket;
+import com.liferay.osb.spring.boot.client.zendesk.search.SearchHits;
+import com.liferay.osb.spring.boot.client.zendesk.search.ZendeskTicketQuery;
+import com.liferay.osb.spring.boot.client.zendesk.service.ZendeskService;
+import com.liferay.portal.kernel.security.auth.PrincipalException;
+import com.liferay.portal.kernel.util.GetterUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Jenny Chen
+ */
+@RestController
+public class BusinessEventRestController extends BaseRestController {
+
+	@RequestMapping(
+		method = RequestMethod.GET,
+		path = "/accounts/{externalReferenceCode}/tickets/{ticketId}"
+	)
+	public ResponseEntity<String> getZendeskTicket(
+		@AuthenticationPrincipal Jwt jwt,
+		@PathVariable("externalReferenceCode") String externalReferenceCode,
+		@PathVariable("ticketId") long ticketId) {
+
+		try {
+			if (!_hasAccountEditAccess(jwt, externalReferenceCode)) {
```

Comment:
this permission check isn't sufficient. Anyone can manipulate the ticketId of other accounts if they use their own account ERC